packages/next/build/webpack-config.ts

```diff
@@ -1586,6 +1586,21 @@ export default async function getBaseWebpackConfig(
               } as any,
             ]
           : []),
+        // Alias `next/dynamic` to React.lazy implementation for RSC
+        ...(hasServerComponents
+          ? [
+              {
+                test: codeCondition.test,
+                include: [appDir],
+                resolve: {
+                  alias: {
+                    [require.resolve('next/dynamic')]:
+                      'next/dist/client/components/dynamic',
+                  },
+                },
+              },
+            ]
+          : []),
```

Comment:
Can we do this directly in `resolve.alias`, instead of adding a new rule?
 https://github.com/vercel/next.js/blob/c7ab8314d759c88c620848f472698207d186d98d/packages/next/build/webpack-config.ts#L841-L858